go/consensus/tendermint/apps/registry/query.go

```diff
@@ -71,6 +73,10 @@ func (rq *registryQuerier) Node(ctx context.Context, id signature.PublicKey) (*n
 	return node, nil
 }
 
+func (rq *registryQuerier) NodeByConsensusID(ctx context.Context, id signature.PublicKey) (*node.Node, error) {
+	return rq.state.NodeByConsensusAddress(ctx, []byte(tmcrypto.PublicKeyToTendermint(&id).Address()))
```

Comment:
Slight problem with this is if you only have a "consensus address" (which is an internal Tendermint detail) and want to lookup the node -- it seems this is not possible but was one of the primary drivers for this feature (see #2961).

We could instead expose `NodeByConsensusAddress` that would take a `[]byte` argument and mention in the documentation comment that what a "consensus address" actually is depends on the consensus backend.